### DISCUSSION TITLE
Border large number of files based on dimensions

### DISCUSSION THREAD

**Discussion by DraganDragon5** [ImageMagick/ImageMagick#3957] [Help] [answered]
hello everyone!

i would like to ask you guys for help with something.

i have a lot of images in a folder that need to be processed similarly. i read that you can use a for loop in imagemagick for this.
i need to border all of the images in my folder, based on their dimensions.

for example of how i need them to be bordered, lets say i have 4 images in this folder with dimension:
- 100x100
- 134x142
- 122x116
- 152x114

if one of the dimensions of an image is divisible by the number 4, that dimension will be bordered with 4 pixels. and if one of the dimensions is not divisible by 4, it would be bordered with 3 pixels.

(divisible by 4: width%4 = 0)

so these 4 images would be bordered like this:

-100x100 => -border 4x4 => new dimensions: 108x108
(100 is divisible by 4)

-134x142 => - border 3x3 => new dim. 140x148
(134 and 142 are not divisible by 4)

- 122x116 => -border 3x4 => new dim. 128x124
(122 is not divisible by four, but 116 is)

- 152x114 => -border 4x3 => new dim. 160x120
(152 is disible by 4, 114 is not)

all of the images in my folder have even dimension.

i know i can use a for loop like this to border 3x3 for example:
for %g in ("*.png") do magick convert -bordercolor black -border 3x3 borderedImages\%g

but can i write a for loop that borders my images like i described above?

i hope i explained the problem well.

thank you in advance and have a nice day😊

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform? I assume Windows. If you don't use "magick" in v7, or if you use "magick convert", this won't work.
```
magick -size 300x403 xc: -bordercolor Blue -border %[fx:w%4==0?4:3]x%[fx:h%4==0?4:3] info:

xc: XC 308x409 308x409+0+0 16-bit sRGB 0.000u 0:00.009
```

"magick" commands need an input and an output filename. So a Windows "for" loop could be:
```
for %F in (*.png) do magick %F -bordercolor Blue -border %[fx:w%4==0?4:3]x%[fx:h%4==0?4:3] outdir\%F
```
If you do this in a BAT file, you need to double the percent signs.

**Reply by DraganDragon5** [reply]
Hey!

I have IM v7, Windows platform. The for loop you provided worked like a charm!!! Thank you so so so much! You've saved me hours of time😁
Again, thank you kind wizard and have a wonderful day!!!!!!!
